lib/stm32wba/CMakeLists.txt

```
@@ -85,52 +85,60 @@ if(CONFIG_BT_STM32WBA)
 
-# Selecting the proper version of link layer lib
-if(CONFIG_BT_STM32WBA)
+# Distinguish STM32WBA6x from STM32WBA5x lines
+if(CONFIG_SOC_STM32WBA65XX)
+  set(libname_prefix "WBA6_")
+elseif(CONFIG_SOC_STM32WBA55XX OR SOC_STM32WBA52XX)
+  set(libname_prefix "")
+else()
+  message(FATAL_ERROR "Unsupported SoC: ${CONFIG_SOC}")
+endif()
+
+# Selecting the proper version of link layer lib and its associated configuration header file
+if(CONFIG_BT_STM32WBA AND CONFIG_IEEE802154_STM32WBA)
+  message(STATUS "Wireless stack enabled in concurrent mode (BLE+IEEE 802.15.4)")
   if(BLE_LIB_TYPE STREQUAL "BLE_LIB_BASIC")
-      zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/ble_basic)
+    zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/concurrent/ble_15_4)
+    set(LL_LIB LinkLayer_BLE_Basic_15_4_lib_Zephyr.a)
   elseif(BLE_LIB_TYPE STREQUAL "BLE_LIB_FULL")
-      zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/ble_full)
-  endif()
-  # Checking all the soc variants and not simply relying on board name
-  if(CONFIG_SOC_STM32WBA65XX)
-    message(STATUS "STM32WBA6 link layer lib selected")
-    if (BLE_LIB_TYPE STREQUAL "BLE_LIB_BASIC")
-      set(LL_LIB "WBA6_LinkLayer_BLE_Basic_lib.a")
-    elseif (BLE_LIB_TYPE STREQUAL "BLE_LIB_FULL")
-      set(LL_LIB "WBA6_LinkLayer_BLE_Full_lib.a")
-    endif()
+    zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/concurrent/ble_full_15_4)
+    set(LL_LIB LinkLayer_BLE_Full_15_4_lib_Zephyr.a)
   endif()
-
-  if(CONFIG_SOC_STM32WBA55XX OR SOC_STM32WBA52XX)
-    message(STATUS "STM32WBA5 link layer lib selected")
-    if (BLE_LIB_TYPE STREQUAL "BLE_LIB_BASIC")
-      set(LL_LIB "LinkLayer_BLE_Basic_lib.a")
-    elseif (BLE_LIB_TYPE STREQUAL "BLE_LIB_FULL")
-      set(LL_LIB "LinkLayer_BLE_Full_lib.a")
-    endif()
+elseif(CONFIG_BT_STM32WBA)
+  message(STATUS "Wireless stack enabled in BLE mode")
+  if(BLE_LIB_TYPE STREQUAL "BLE_LIB_BASIC")
+    zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/ble_basic)
+    set(LL_LIB LinkLayer_BLE_Basic_lib.a)
+  elseif(BLE_LIB_TYPE STREQUAL "BLE_LIB_FULL")
+    zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/ble_full)
+    set(LL_LIB LinkLayer_BLE_Full_lib.a)
   endif()
 elseif(CONFIG_IEEE802154_STM32WBA)
+  message(STATUS "Wireless stack enabled in IEEE 802.15.4 mode")
   if(CONFIG_SOC_STM32WBA65XX)
     if(CONFIG_OPENTHREAD)
-      message(STATUS "WBA6_LinkLayer_Thread_lib_Zephyr.a lib selected")
-      set(LL_LIB "WBA6_LinkLayer_Thread_lib_Zephyr.a")
+      zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/thread)
+      set(LL_LIB LinkLayer_Thread_lib_Zephyr.a)
     else()
-      message(STATUS "WBA6_LinkLayer15_4_Zephyr.a lib selected")
-      set(LL_LIB "WBA6_LinkLayer15_4_Zephyr.a")
+      zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/ieee_15_4_basic)
+      set(LL_LIB LinkLayer15_4_Zephyr.a)
     endif()
   endif()
 endif()
 
```

And a common `zephyr_include_directories()` used:
```suggestion
zephyr_include_directories(STM32_WPAN/link_layer/ll_cmd_lib/config/${STACK_CFG_NAME})
```